### PR TITLE
chore: fix link to doc webpage

### PR DESCRIPTION
**Pull request by clason**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,3 @@
 Generates doc and reports for neovim.
 
-Serves them at www.neovim.io/doc afterwards.
+Serves them at www.neovim.io/doc/user/ afterwards.
```
Comment (rktjmp) [inline README.md:3]: "Serves *them*" or "Serves *the documentation*" :)

**Comment by jamessan** [comment]
I'd rather fix the way the site is generated so we don't break existing links.

**Comment by clason** [comment]
I don't understand -- I'm not changing anything code wise, just correcting the (broken, resulting in 404) link in the README (since that came up several times now on Matrix)? Other (correct) links will continue working, of course.

(Agreed on the need for general fixes, of course.)

**Comment by jamessan** [comment]
> just correcting the (broken, resulting in 404) link in the README

That broken link is exactly the problem.  It used to work, but then the way the site was generated changed and _caused_ the link to be broken.  I'd rather undo that breakage, fixing _all_ the broken links to `/doc/`, rather than changing our _one_ broken link.

**Pull request closed without merging**